### ISSUE TITLE
Document file structure of naked code stream

### ISSUE THREAD

**Issue by StefanOltmann** [libjxl/libjxl#3129]
I understand that for the naked bytestream there is no EXIF or XMP metadata. But there must be at least image size and rotation metadata included.

My [metadata lib](https://github.com/Ashampoo/kim) should at least identify the file as JPEG XL and report these values.

Can you document the structure of JXL in a more schematic way?

In the docs you refer to C implementation code, but I’m not too familiar with that syntax. Something on a higher level would be helpful.

**Comment by Traneptora** [comment]
The specification of the codestream is publically available on the ISO website as ISO/IEC 18181-1, but it is not free. The best you can get for free is to see how FOSS implementations handle it. libjxl code is a bit hard to parse, but other FOSS decoders exist, notably jxlatte in Java and ixl-oxide in Rust. j40 and ffmpeg both have jxl header parsers but they are in C.

**Comment by StefanOltmann** [comment]
Thanks, will take a look into jxlatte.
